agenthub/codeact_agent/codeact_agent.py

```diff
@@ -129,6 +131,8 @@ def get_action_message(self, action: Action) -> dict[str, str] | None:
             or isinstance(action, CmdRunAction)
             or isinstance(action, IPythonRunCellAction)
             or isinstance(action, MessageAction)
+            or isinstance(action, AgentFinishAction)
+            and action.source == 'agent'
```

Comment:
no parenthesis around all `isinstance`, since they're `or'?

Reply:
Hah, that got my eye too. Only for AgentFinishAction it is necessary, and `and` has precedence. The rest (MessageAction) could be `user` or `agent` and we want to insert both of those

Reply:
For readability a parenthesis could avoid misunderstanding, imho.